test/evergreen/code_change_report/code_change_report.py

```diff
@@ -0,0 +1,210 @@
+    return branch_info
+
+
+def get_non_zero_count(value_list: list):
+    non_zero_count = 0
+    for value in value_list:
+        if value != 0:
+            non_zero_count += 1
+    return non_zero_count
+
+
+def get_html_colour(count: int, of: int):
+    colour = ""
+    if count == 0:
+        colour = "Red"
+    elif count == of:
+        colour = "LightGreen"
+    else:
+        colour = "Orange"
+
+    return colour
+
+
+def generate_file_info_as_html_text(file: str, file_info: dict, verbose: bool):
+    report = list()
+
+    if file.startswith("src/"):
+        report.append("<h3>File: {}</h3>\n".format(html.escape(file, quote=True)))
+
+        for hunk in file_info:
+            lines = hunk['lines']
+            report.append("<table>\n")
+            report.append("  <tr>\n")
+            report.append("    <th>Count</th>\n")
+            report.append("    <th>Branches</th>\n")
+            report.append("    <th>=+-</th>\n")
+            report.append("    <th></th>\n")
+            report.append("  </tr>\n")
+            for line in lines:
+                print("line: " + str(line))
+                new_lineno = line['new_lineno']
+                old_lineno = line['old_lineno']
+                content = line['content']
+                code_colour = "White"
+                plus_minus = "="
+                strikethrough = False
+                if new_lineno > 0 > old_lineno:
+                    plus_minus = "+"
+                if old_lineno > 0 > new_lineno:
+                    plus_minus = "-"
+                    strikethrough = True
+                count_str = ""
+                if 'count' in line:
+                    count = line['count']
+                    if count >= 0 and content != '\n':
+                        count_str = str(count)
+                        code_colour = get_html_colour(count, count)
+                report.append("  <tr>\n")
+                report.append("    <td>{}</td>\n".format(count_str))
+                if 'branches' in line:
+                    branches = line['branches']
+                    branch_info = get_branch_info(branches=branches)
+                    num_branches = len(branches)
+                    if num_branches > 0:
+                        non_zero_count = get_non_zero_count(branch_info)
+                        code_colour = get_html_colour(non_zero_count, num_branches)
+                        report.append("    <td>\n")
+                        report.append("      <details>\n")
+                        report.append("      <summary>\n")
+                        report.append("      {} of {}\n".format(non_zero_count, num_branches))
+                        meter_high = len(branch_info) * 0.999
+                        report.append(
+                            "      <meter value=\"{}\" optimum=\"{}\" max=\"{}\" high=\"{}\" low=\"{}\"></meter>".
+                            format(non_zero_count, num_branches, num_branches, meter_high, meter_high))
```

Comment:
I'm not familiar with meters, but should this be
```suggestion
                            format(non_zero_count, num_branches, num_branches, meter_high, meter_low))
```
?

Reply:
That's deliberate. I wanted anything less than the top value to make the scale red, and the way to do that, apparently, is to set the low value to the high value/maximum. I'll add a comment explaining that though, as that would seem valuable.

Reply:
I'll also create a meter_low value and assign it the value of meter_high as I think that will make it clearer.